shared/components/modals/OfferModal/AddOffer/AddOffer.js

```diff
@@ -400,32 +418,34 @@ export default class AddOffer extends Component {
   render() {
     const { currencies, tokenItems, addSelectedItems } = this.props
     const { exchangeRate, buyAmount, sellAmount, buyCurrency, sellCurrency, minimalestAmountForSell, minimalestAmountForBuy,
-      balance, isBuyFieldInteger, isSellFieldInteger, ethBalance, manualRate, isPartial, isToken } = this.state
+      balance, isBuyFieldInteger, isSellFieldInteger, ethBalance, manualRate, isPartial, isTokenSell, isTokenBuy } = this.state
     const linked = Link.all(this, 'exchangeRate', 'buyAmount', 'sellAmount')
     const minAmountSell = coinsWithDynamicFee.includes(sellCurrency) ? minimalestAmountForSell : minAmountOffer[sellCurrency]
     const minAmountBuy = coinsWithDynamicFee.includes(buyCurrency) ? minimalestAmountForBuy : minAmountOffer[buyCurrency]
 
-    const minimalAmountSell = !isToken ? Math.floor(minAmountSell * 1e6) / 1e6 : 0
-    const minimalAmountBuy = !isToken ? Math.floor(minAmountBuy * 1e6) / 1e6 : 0
+    const minimalAmountSell = !isTokenSell ? Math.floor(minAmountSell * 1e6) / 1e6 : 0
```

Comment:
То есть, если токен, то минималка будет 0? Я могу свопнуться на 1е-20 или 1е-22 токена, хотя его decimals максимум 18, и следовательно минималка реальная 1е-18

Вынести этот метод в `heplpers.ethToken.estimateMinAmount({ name } = {})`

![screenshot_20190213_181847](https://user-images.githubusercontent.com/43396375/52706287-bcdfd700-2fd0-11e9-86c5-199a740affb8.png)

Добавь в него `const decimals = constants.tokenDecimals[name.toLowerCase()]` вместо `currentDecimals`
А лучше переименуй constants.tokenDecimals -> constants.decimals, так как там не только токены. Этот конфиг используется только в двух модалках, включая эту.

И присваивай minAmount токенам по их имени. 

`Math.floor(minAmountSell * 1e6) / 1e6` лучше заменить на `Number(x).dp(decimals, 6)`

```suggestion
    const minimalAmountSell = !isTokenSell 
      ? BigNumber(minAmountSell).dp(constants.tokenDecimals[sellCurrency.toLowerCase()], BigNumber.ROUND_DOWN).toSting()
      : helpers.ethToken.estimateMinAmount({  name: sellCurrency })
```

Reply:
> `Math.floor(minAmountSell * 1e6) / 1e6` лучше заменить на `Number(x).dp(decimals, 6)`

BigNumber?

Reply:
> > `Math.floor(minAmountSell * 1e6) / 1e6` лучше заменить на `Number(x).dp(decimals, 6)`
> 
> BigNumber?

Конечно! Опечатка просто